src/overlays/actors/ovl_Lamp_Light/ef_lamp_light.c

```diff
@@ -19,21 +25,114 @@ ActorProfile Lamp_Light_Profile = {
     /* */ Ef_Lamp_Light_actor_ct,
     /* */ Ef_Lamp_Light_actor_dt,
     /* */ Ef_Lamp_Light_actor_move,
-    /* */ (ActorFunc)none_proc1,
+    /* */ (void*)none_proc1,
     /* */ NULL,
 };
-#endif
+static s16 D_80967884_jp[4] = { 2, 2, 1, 0 };
+static s16 D_8096788C_jp[4] = { 0x10, 0x10, 8, 0 };
+static s16* D_80967894_jp[2] = { D_80967884_jp, D_8096788C_jp };
+static s16 D_8096789C_jp[4] = { 0, 0, 0, 0 };
+static s16 D_809678A4_jp[4] = { 0xC8, 0xC8, 0x96, 0 };
+static s16* D_809678AC_jp[2] = { D_8096789C_jp, D_809678A4_jp };
+
+void Ef_Lamp_Light_actor_ct(Actor* thisx, Game_Play* game_play) {
+    s32 pad UNUSED;
+    Lamp_Light* this = (Lamp_Light*)thisx;
+
+    game_play->kankyo.unk_1C = &this->unk_178;
+    Light_diffuse_ct(&this->unk_178, 0, 0x50, 0, (u8)0, (u8)0, (u8)0);
```

Comment:
Are the casts required?
```suggestion
    Light_diffuse_ct(&this->unk_178, 0, 0x50, 0, 0, 0, 0);
```